### PR TITLE
Remove empty lines surrounding the lone comments in struct

### PR DESCRIPTION
**Pull request by ebati**
Remove empty lines surrounding comments in struct/interfaces. In short, converts
```
type S struct {

    // comment

}
```
to
```
type S struct {
    // comment
}
```

### CODE REVIEW

File: format/format.go [hunk 1]
```diff
@@ -584,13 +584,34 @@ func (f *fumpter) applyPre(c *astutil.Cursor) {
...
+			var bodyPos, bodyEnd token.Pos
+			if numFields > 0 {
+				bodyPos = node.List[0].Pos()
+				bodyEnd = node.List[len(node.List)-1].End()
+			}
+			if len(comments) > 0 {
+				if pos := comments[0].Pos(); !bodyPos.IsValid() || pos < bodyPos {
+					bodyPos = pos
+				}
+				if pos := comments[len(comments)-1].End(); !bodyPos.IsValid() || pos > bodyEnd {
```
Comment (mvdan) [inline format/format.go:609]: I can tell that you thought about the edge cases here :)

**Review by mvdan — APPROVED** [review]
Thanks! Nicely done. More tests for e.g. combining fields and comments would be nice, but not hugely important right now.
